lib/liquid/standardfilters.rb

```diff
@@ -125,7 +125,7 @@ def sort(input, property = nil)
       elsif ary.empty? # The next two cases assume a non-empty array.
         []
       elsif ary.first.respond_to?(:[]) && !ary.first[property].nil?
-        ary.sort { |a, b| a[property] <=> b[property] }
+        ary.sort { |a, b| a[property] && b[property] ? a[property] <=> b[property] : a[property] ? -1 : 1 }
```

Comment:
It would also be nice to avoid evaluating `[]` multiple times. 👍  to breaking it up into multiple lines, and also caching the value of `[property]`.